Quizzes!!! (#25)
fhdsl/Tools_for_Reproducible_Workflows_in_R: quizzes/quiz_ch3.md

{quiz, id: components, attempts: 10}

## Components of reproducibility

? What does "reproducibility on a continuum" mean?

a) Reproducibility that can only be achieved with perfect execution
B) Reproducibility that needs iterative work to improve
c) Reproducibility that is dependent on the researcher's expertise
d) Reproducibility that is achievable only for certain types of analyses

? Which of the following is not a component of reproducibility?

a) Transparency
b) Accessibility
c) Consistency
D) Efficiency

? What does "open source" mean in the context of reproducibility?

> A)  Making data and code available for anyone to access and modify

Reviewer: ```suggestion
A) Making data and code available for anyone to access and suggest modifications
```

Reviewer: "and modify" sounds like random people might modify your code - maybe we mean to access and use? just wondering if we could phrase it a bit differently. I know you mean code review, but just thinking for people new to the subject that this might sound concerning

Author: How's this? @carriewright11

Reviewer: looks good!

b) Providing data and code upon request by email
c) Publishing data and code behind a paywall
d) Storing data and code on a personal computer

? Why is data sharing important in promoting the open source-ness of an analysis?

a) It makes the analysis more complex and difficult to execute
b) It helps others to be able to publish conclusions about our data
C) It helps others be able to truly reproduce the analysis by using the same data
d) It exposes personal identifiable information to the public

? True or False - if your data has PHI or PII you cannot make your analysis open source or reproducible.

a) TRUE - There is no methods that will allow you to share your methods and data in a safe way.

> B) FALSE! - These are not mutually exclusive goals! but will take a bit of thoughtful planning and using controlled access methods so that all individuals' data are well protected!

Reviewer: ```suggestion
B) FALSE! - These are not mutually exclusive goals! Some thoughtful planning and use of controlled access methods can ensure that all individuals' data are well protected!
```
